### DISCUSSION TITLE
How to do integration tests in CI of a service which uses Mongo encryption and GCP KMS ?

### DISCUSSION THREAD

**Discussion by theliuk** [mia-platform/community#392] [General]
### Description

We are developing a service in python which uses these dependencies:

1. MongoDB
2. GCP KMS
3. Redis

We are wondering what is the best way to write down automatic integration tests that can be run in the Gitlab CI (where the service is built).

We've experience with "simple" integration tests where the service depends only on Mongo (without encryption); in this case we know how to setup a MongoDB instance with a docker image inside the runtime of the CI.
In this case, however, the encryption requires MongoDB Enterprise or Atlas.

How were integration tests made for the `crud-service` to test the encryption features? 
Do you suggest to have such integration tests running in the CI, or is it better to run them manually in a test environment?

Thank you,

FYI @alexBonadei

**Comment by theliuk** [comment]
Help us plz 😢

**Reply by danibix95** [reply]
Hi! Have you already had a look at the Github actions of [`CRUD Service`](https://github.com/mia-platform/crud-service), in particular the one for [testing](https://github.com/mia-platform/crud-service/blob/main/.github/workflows/tests.yml) the service?

From there it should be possible to take inspiration for writing the Gitlab pipeline.

The gist is that it is necessary to install the encryption library on the test runner. The library version may vary depending on your system, database version and driver version.
